content/blog/2018-09-14-smapr.md

```diff
@@ -0,0 +1,637 @@
+## 1 /home/rstudio/.cache/smap
+## 2 /home/rstudio/.cache/smap
+## 3 /home/rstudio/.cache/smap
+## 4 /home/rstudio/.cache/smap
+## 5 /home/rstudio/.cache/smap
+## 6 /home/rstudio/.cache/smap
+```
+
+Next, we can list the contents of the files we downloaded.
+These are HDF5 files, so each file contains multiple datasets.
+For this data product, a soil moisture data set is named `Soil_Moisture_Retrieval_Data_AM/soil_moisture`, and we can use the `extract_smap` function to generate rasters from this dataset (see `list_smap` for a list of all datasets contained in any file):
+
+
+```r
+sm_raster <- extract_smap(downloads, name = "Soil_Moisture_Retrieval_Data_AM/soil_moisture")
+sm_raster
+```
+
+```
+## class       : RasterBrick
+## dimensions  : 406, 964, 391384, 6  (nrow, ncol, ncell, nlayers)
+## resolution  : 36032.22, 36032.22  (x, y)
+## extent      : -17367530, 17367530, -7314540, 7314540  (xmin, xmax, ymin, ymax)
+## coord. ref. : +proj=cea +lon_0=0 +lat_ts=30 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs +ellps=WGS84 +towgs84=0,0,0
+## data source : /home/rstudio/.cache/smap/tmp.tif
+## names       : SMAP_L3_S//R16010_001, SMAP_L3_S//R16010_001, SMAP_L3_S//R16010_001, SMAP_L3_S//R16010_001, SMAP_L3_S//R16010_001, SMAP_L3_S//R16010_001
+## min values  :                  0.02,                  0.02,                  0.02,                  0.02,                  0.02,                  0.02
+## max values  :             0.9122642,             0.9390723,             0.9531010,             0.9122642,             0.9531010,             0.9390723
+```
+
+Let's plot the RasterBrick to see what the soil moisture data look like:
+
+
+```r
+levelplot(sm_raster)
+```
+
+![](/img/blog-images/2018-09-14-smapr/plot-smap-1.png)<!-- -->
+
+The striping occurs because of the orbit of the SMAP satellite.
+If we were interested in some weekly measure of soil moisture, we could simply average across days to get a more continuous picture of soil moisture:
+
+
+```r
+weekly_sm <- mean(sm_raster, na.rm = TRUE)
+plot(weekly_sm)
+```
+
+![](/img/blog-images/2018-09-14-smapr/get-weekly-mean-1.png)<!-- -->
+
+We'll want to do that set of operations a bunch of times for this case study, so we'll write a little function that takes a date range and returns a raster averaging over dates:
+
+
+```r
+average_smap <- function(date_range) {
+  mean_sm <- find_smap('SPL3SMP', dates = date_range, version = 5) %>%
+    download_smap(overwrite = FALSE) %>%
+    extract_smap(name = "Soil_Moisture_Retrieval_Data_AM/soil_moisture") %>%
+    mean(na.rm = TRUE)
+  mean_sm
+}
+```
+
+Notice that spatial coverage is still not 100%.
+Notably, places where the satellite missed and frozen regions in the arctic have `NA` values.
+
+## Getting global precipitation data with rnoaa
+
+We can better interpret soil moisture if we know about precipitation, and the NOAA Climate Prediction Center (CPC) has global precipitation data that are a cinch to get with the `cpc_prcp` function.
```

Comment:
reading to this point and your narrative style is fantastic - good clear explanations of what you're doing and what is represented in the figs

Reply:
Thanks!